File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> group;
        unordered_map<string, vector<string>> m;
        for(int i=0; i<strs.size(); i++){
            string x = strs[i];
            string temp = x;
            sort(temp.begin(), temp.end());

            m[temp].push_back(x);
        }

        for(auto &it:m){
            group.push_back(it.second);
        }

        return group;
    }
};

File: 49-group-anagrams/group-anagrams.go
func groupAnagrams(strs []string) [][]string {
    group := make(map[string][]string)

    for i := range strs {
        str := []byte(strs[i])

        sort.Slice(str, func (i, j int) bool {
            return str[i] < str[j]
        })

        group[string(str)] = append(group[string(str)], strs[i])
    }

    var ans [][]string
    for _, val := range group {
        ans = append(ans, val)
    }

    return ans
}